FinniversKit/Sources/Fullscreen/AdConfirmationView/Views/AdConfirmationLinkView.swift

```diff
@@ -0,0 +1,129 @@
+    func adConfirmationLinkView(_ view: AdConfirmationLinkView, buttonWasTapped sender: UIButton)
+}
+
+public class AdConfirmationLinkView: UIView {
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel(withAutoLayout: true)
+        label.text = model.title
+        label.font = .bodyStrong
+        label.textColor = .textPrimary
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var bodyLabel: UILabel = {
+        let label = UILabel(withAutoLayout: true)
+        label.font = .body
+        label.text = model.body
+        label.textColor = .textPrimary
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var linkButton: Button = {
+        let button = Button(style: .default, size: .small, withAutoLayout: true)
+        button.setTitle(model.linkTitle, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(buttonWasTapped(_:)), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var linkIcon: UIImageView = {
+        let imageView = UIImageView(withAutoLayout: true)
+        imageView.image = UIImage(named: .webview)
+        return imageView
+    }()
+
+    private var model: AdConfirmationLinkViewModel
+    private weak var delegate: AdConfirmationLinkViewDelegate?
+
+    init(model: AdConfirmationLinkViewModel, delegate: AdConfirmationLinkViewDelegate?) {
+        self.model = model
+        self.delegate = delegate
+
+        super.init(frame: .zero)
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = .bgPrimary
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setup() {
+        let iconSize: CGFloat = 12
+        let margins = UIEdgeInsets(
+            top: .spacingS,
+            leading: .spacingM,
+            bottom: .spacingS,
+            trailing: .spacingM + iconSize
+        )
+
+        linkButton.style = linkButton.style.overrideStyle(margins: margins)
+        linkButton.addSubview(linkIcon)
+
+        NSLayoutConstraint.activate([
+            linkIcon.widthAnchor.constraint(equalToConstant: iconSize),
+            linkIcon.heightAnchor.constraint(equalToConstant: iconSize),
+            linkIcon.trailingAnchor.constraint(equalTo: linkButton.trailingAnchor, constant: -.spacingS - linkButton.style.borderWidth),
+            linkIcon.centerYAnchor.constraint(equalTo: linkButton.centerYAnchor)
+        ])
+
+        addSubview(linkButton)
+        NSLayoutConstraint.activate([
+            linkButton.centerXAnchor.constraint(equalTo: centerXAnchor),
+            bottomAnchor.constraint(equalTo: linkButton.bottomAnchor)
+        ])
+
+        let maxWidth: CGFloat = 640 - .spacingXL
```

Comment:
Why this specific width :P?

Reply:
👍  I'll get rid of it. 😄 It was mostly an idea to limit the possibility of extremely wide texts on iPad in landscape, but it's not really a likely scenario for this element at all.